ql/src/java/org/apache/hadoop/hive/ql/udf/generic/AbstractGenericUDFArrayBase.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.udf.generic;
+
+import org.apache.hadoop.hive.ql.exec.UDFArgumentException;
+import org.apache.hadoop.hive.ql.exec.UDFArgumentTypeException;
+import org.apache.hadoop.hive.serde.serdeConstants;
+
+import org.apache.hadoop.hive.serde2.objectinspector.ListObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorConverters;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorConverters.Converter;
+import org.apache.hadoop.hive.serde2.objectinspector.PrimitiveObjectInspector;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Abstract GenericUDF for array functions
+ */
+
+public abstract class AbstractGenericUDFArrayBase extends GenericUDF {
+
+    static final int ARRAY_IDX = 0;
+    static final int ARRAY2_IDX = 1;
+    static final int START_IDX = 1;
+    static final int LENGTH_IDX = 2;
+    static final int SEPARATOR_IDX = 1;
+    static final int REPLACE_NULL_IDX = 2;
+
+    private final int minArgCount;
+    private final int maxArgCount;
+    private final ObjectInspector.Category outputCategory;
+
+    private final FUNC_NAMES funcName;
+
+    transient ListObjectInspector arrayOI;
+    transient ObjectInspector[] argumentOIs;
+
+    transient Converter converter;
+
+    public AbstractGenericUDFArrayBase(FUNC_NAMES funcName, int minArgCount, int maxArgCount, ObjectInspector.Category outputCategory) {
+        this.funcName = funcName;
+        this.minArgCount = minArgCount;
+        this.maxArgCount = maxArgCount;
+        this.outputCategory = outputCategory;
+    }
+
+    enum FUNC_NAMES {
+        ARRAY_MAX, ARRAY_MIN, ARRAY_DISTINCT, ARRAY_SLICE, ARRAY_JOIN, ARRAY_EXCEPT, ARRAY_INTERSECT
+    }
+
+    @Override
+    public ObjectInspector initialize(ObjectInspector[] arguments)
+            throws UDFArgumentException {
```

Comment:
Better to call `initialize` in sub-classes and maintain common function like `checkArgsSize ` and `checkArgCategory ` here. This will remove the need of maintaining function types (enums) in the code.

Reply:
Removed ENUM